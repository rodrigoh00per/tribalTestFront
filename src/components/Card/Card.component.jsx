import React, { useMemo } from "react";
import { DeleteIcon, EditIcon } from "../../styles/Icons.styles";
import { getMask } from "../Forms/Mask.utils";
import {
  ContainerCard,
  Email,
  IconsCard,
  Phone,
  Subtitle,
  TextsBottomCard,
  TextsTopCard,
  Title,
} from "./Card.styles";

const Card = ({
  id,
  title = "",
  subtitle = "",
  phone = "",
  email = "",
  withDelete,
  withEdit,
  onClickAction = () => {},
}) => {
  const phoneFormated = useMemo(
    () => getMask("phone")(phone),
    [phone]
  );

  return (
    <ContainerCard>
      <TextsTopCard>
        <Title>{title}</Title>
        <Subtitle>{subtitle}</Subtitle>
      </TextsTopCard>
      <IconsCard>
        {withEdit ? (
          <EditIcon
            onClick={(e) => {
              e.stopPropagation();
              onClickAction("edit", id, title);
            }}
          />
        ) : null}
        {withDelete ? (
          <DeleteIcon
            onClick={(e) => {
              e.stopPropagation();
              onClickAction("delete", id, title);
            }}
          />
        ) : null}
      </IconsCard>
      <TextsBottomCard>
        <Phone>{phoneFormated}</Phone>
        <Email>{email}</Email>
      </TextsBottomCard>
    </ContainerCard>
  );
};

export default React.memo(Card);
