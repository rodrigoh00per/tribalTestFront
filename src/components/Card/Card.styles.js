import styled from "styled-components";


export const ContainerCard = styled.div`
    margin-bottom: 20px;
    width: 304px;
    height: 158px;
    display: grid;
    border: 1px solid #C8C8CC;
    grid-template-columns: 240px 64px;
    grid-template-rows: 79px 79px;
    border-radius: 12px;
`;

export const TextsTopCard = styled.div`
    grid-row: 1;
    grid-column: 1;
    border-bottom: 1px solid #EFEEEE;
    padding: 0 0 0 15px;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
`;
export const IconsCard = styled.div`
    grid-row: 1;
    grid-column: 2;
    border-bottom: 1px solid #EFEEEE;
    padding-top: 10px;
`;

export const TextsBottomCard = styled.div`
    grid-row: 2;
    grid-column: 1 / 3;
    padding:0 0 0 15px;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
`;
export const Title = styled.label`
    font-weight: 600; 
    fontSize: 16px;
    color: black;
`;
export const Subtitle = styled.label`
    font-weight: normal;
    font-size: 14px; 
    color: #8D929A; 
`;

export const Phone = styled.label`
    font-weight: 600; 
    font-size: 14px;
    color: black;

`;
export const Email = styled.label`
    font-weight: 600;
    font-size: 14px; 
    color: black;

`;