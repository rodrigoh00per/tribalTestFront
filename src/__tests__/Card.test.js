import React from "react";
import Card from "../components/Card/Card.component";
import { shallow, mount } from "enzyme";

describe("----TEST OF THE CARD---", () => {

    it("take a snapshot of the card component", () => {
        expect(shallow(<Card />)).toMatchSnapshot();
    })

    it("expect that receive a title", () => {
        const mockInfo = { title: "test title" };

        const wrapper = mount(<Card withDelete={true} withEdit={true} title={mockInfo.title} />);

        expect(wrapper.props().title).toEqual("test title");
    });

});
