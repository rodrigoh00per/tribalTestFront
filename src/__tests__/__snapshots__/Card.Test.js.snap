// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----TEST OF THE CARD--- take a snapshot of the card component 1`] = `ShallowWrapper {}`;
